blocks/filter-group/filter-group.js

```diff
@@ -1,42 +1,193 @@
+import { dataStore } from '../../scripts/dataStore.js';
+
+/**
+ * Update selected state of filter button.
+ * @param {HTMLButtonElement} buttonElement 
+ * @param {boolean} newSelected New selected state
+ */
+const updateFilter = (buttonElement, newSelected = false) => {
+    buttonElement.classList.toggle('filter-group__button--selected', newSelected);
+    buttonElement.setAttribute('aria-pressed', newSelected ? 'true' : 'false');
+};
+
+/**
+ * Get an array containing the text value of all selected filters.
+ * When "All" is selected, this returns an empty array.
+ * @param {Element} parentElement 
+ * @returns {string[]}
+ */
+const getCurrentFiltersArray = (parentElement) => Array.from(
+    parentElement.querySelectorAll('.filter-group__button:not(:first-of-type)[aria-pressed="true"]'), btn => btn.textContent
+) || [];
+
+/**
+ * TO-DO: remove this when ideas page content and functionality is finalized.
```

Comment:
```suggestion
 * TODO: remove this when ideas page content and functionality is finalized.
```

if you don't put a dash there it'll activate my advanced syntax ide highlighting 👼